src/components/record/related-table-actions.tsx

```diff
@@ -496,50 +603,293 @@ const RelatedTableActions = ({
+                Explore more {currentTable} records related to this {mainTable}.
+              </span>
+            }
+          >
+            <a
+              className='chaise-btn chaise-btn-tertiary more-results-link dropdown-button'
+              href={exploreLink}
+              onClick={onExplore}
+            >
+              <span className='chaise-btn-icon fa fa-pencil'></span>
+              <span>Bulk Edit</span>
+            </a>
+          </ChaiseTooltip>
+        );
+      case 'Link Records':
+      case 'Add Records':
+        tooltip = createBtnTooltip;
+        break;
+      case 'Unlink records':
+        tooltip = unlinkBtnTooltip;
+        break;
+      case 'Edit mode':
+      case 'Custom mode':
+      case 'Table mode':
+        return renderCustomModeBtn(true);
+    }
+
+    return (
+      <ChaiseTooltip placement='top' tooltip={tooltip}>
+        <button
+          key={`button-${index}`}
+          className={'chaise-btn chaise-btn-tertiary dropdown-button'}
+        >
+          <span className={spanClass}></span>
+          <span className='dropdown-button-text'>{buttonText}</span>
+        </button>
+      </ChaiseTooltip>
+    );
+  };
+  /*
+   * This hook is to push the buttons to the dropdown if it exceeds the container width.
+   * We are using ResizeSensor to listen to the resize event.
+   */
+  useLayoutEffect(() => {
+    const mainnav = document.querySelector('#navheader') as HTMLElement;
+    if (!buttonRef.current) return;
+
+    const calculateButtons = () => {
+      setIsMobile(window.innerWidth < 600);
+      let buttons: any = [];
+      if (containerRef.current) {
+        const buttonContainer = buttonRef.current;
+
+        const containerWidth = buttonContainer.getBoundingClientRect().width;
+        buttons = Array.from(buttonContainer.getElementsByClassName('chaise-btn'));
+
+        const tableHeaderButtonsToAddToDropdown: any = [];
+        let buttonLeftOffset = 0;
+        let totalWidth = 0;
+
+        if (isMobile) {
+          setButtonsInDropdown(buttons);
+          buttonContainer.style.display = 'none';
+          return;
+        } else {
+          buttonContainer.style.display = 'block';
+          setButtonsInDropdown([]);
+          buttons.forEach((button: any, index: number) => {
+            const buttonWidth = button.getBoundingClientRect().width;
+            if (index === 0) {
+              buttonLeftOffset =
+                button.getBoundingClientRect().left - buttonContainer.getBoundingClientRect().left;
+            }
+
+            if (totalWidth + buttonLeftOffset + buttonWidth <= containerWidth) {
+              totalWidth += buttonWidth;
+            } else {
+              tableHeaderButtonsToAddToDropdown.push(button);
+            }
+          });
+          setButtonsInDropdown(tableHeaderButtonsToAddToDropdown);
+        }
+      }
+    };
+    calculateButtons();
+
+    const mainNavResizeSensor = new ResizeSensor(mainnav as Element, () => {
```

Comment:
This doesn't make any sense. We want to have the `ResizeSensor` on an element where its height/width change will affect the buttons. So it should be on either the parent of the button containers, or some other elements that affect the container. Let's use `.main-container` for now.